dataservice_publisher/app.py

```diff
@@ -44,6 +51,109 @@ async def authenticated(request: web.Request) -> bool:
     return False
 
 
+class MimeType:
+    """Class for handling mime types."""
+
+    def __init__(self, content_type: str, q: float = 1.0) -> None:
+        """Initialize the mime type."""
+        self.content_type = content_type
+        self.q = q
+
+    def __eq__(self, other: Any) -> bool:  # pragma: no cover
+        """Compare two mime types."""
+        if isinstance(other, str):
+            return self.content_type == other
+        if isinstance(other, MimeType):
+            return self.content_type == other.content_type
+        return False
+
+    def __str__(self) -> str:
+        """Return the mime type as a string."""
+        return self.content_type
+
+
+async def prepare_mime_types(accept_mime_types: List[str]) -> List[str]:
+    """Prepare the accept mime types and sort on q-parameter."""
+    logging.debug(f"Prcoessing accept mime types: {accept_mime_types}")
+    # Assign q-parameter:
+    accept_mime_types_sorted: List[MimeType] = []
+
+    for accept_mime_type in accept_mime_types:
+        mime_type_split = accept_mime_type.split(";")
+        mime_type = MimeType(mime_type_split[0])
+
+        # If q-parameter is present, assign it:
+        for mime_type_part in mime_type_split[1:]:
+            if mime_type_part.startswith("q="):
+                mime_type.q = float(mime_type_part.split("=")[1])
+
+        accept_mime_types_sorted.append(mime_type)
+
+    # Adjust q-parameters with regard to specificity:
+    # Highest q-parameter is the most specific:
+    for mime_type in accept_mime_types_sorted:
+        logging.debug(f"Ajusting q-parameter for mime type: {mime_type}")
+        if mime_type.content_type.split("/")[0] == "*":
+            pass
+        elif mime_type.content_type.split("/")[1] == "*":
+            mime_type.q = float(mime_type.q) + 0.01
+        else:
+            mime_type.q = float(mime_type.q) + 0.02
```

Comment:
```suggestion
            mime_type.q = float(mime_type.q) + 0.0001
        else:
            mime_type.q = float(mime_type.q) + 0.0002
```
> in the range 0 through 1, where 0.001 is the least preferred and 1 is the most preferred

https://httpwg.org/specs/rfc7231.html#quality.values